src/index.ts

```diff
@@ -1 +1,77 @@
-console.log('Hello World');
+import {Environment, ErrorData, ErrorReport, ErrorStackTrace, ErrorTrace} from "./models";
+import {generateUUIDv4} from "./utils";
+
+const reportVersion = 1;
+
+let sessionId: string;
+
+const stackTraceLineRegex = /^\s+at\s(\S+)\s\((.+):(\d+):(\d+)\)$/;
+
+export default function InitializeErrorTracker() {
+  sessionId = generateUUIDv4();
+
+  window.addEventListener("error", (error: ErrorEvent) => {
+    console.log(prepareReport(error));
+  });
+
+  console.debug('Error Tracker Initialized!');
+}
+
+function prepareReport(error: ErrorEvent): ErrorReport {
+  return {
+    version: reportVersion,
+    environment: collectEnvironmentData(),
+    errorData: parseError(error),
+    language: 'javascript',
+    sessionId: sessionId
+  };
+}
+
+function parseError(error: ErrorEvent): ErrorData {
+  return {
+    message: error.message,
+    date: new Date(),
+    class: error.error.name,
+    trace: parseErrorTrace(error.error.stack),
+    handled: error.defaultPrevented,
+    timeSinceOrigin: error.timeStamp
+  };
+}
+
+function parseErrorTrace(stack: string): ErrorTrace {
+  return {
+    stack: parseErrorStackTrace(stack),
+    string: stack
+  };
+}
+
+function parseErrorStackTrace(stack: string): ErrorStackTrace[] {
+  return stack.split('\n').slice(1).map((line) => {
+    const parsed = stackTraceLineRegex.exec(line);
+    if (parsed) {
+      return {
+        row: parseInt(parsed[3], 10),
```

Comment:
Check length of array before using it

Reply:
The array comes from a regex, it is bounded by start and end, so it cannot have fewer than 5 elements.